engine/crates/engine/src/registry/resolvers/federation.rs

```diff
@@ -0,0 +1,90 @@
+use futures_util::future::join_all;
+use serde::Deserialize;
+use serde_json::{json, Value};
+
+use crate::{
+    registry::{federation::FederationResolver, variables::VariableResolveDefinition, NamedType},
+    Context, ContextExt, ContextField, Error,
+};
+
+use super::{dynamo_querying::DynamoResolver, ResolvedValue, ResolverContext};
+
+#[derive(serde::Deserialize)]
+struct Representation {
+    #[serde(rename = "__typename")]
+    ty: NamedType<'static>,
+    #[serde(flatten)]
+    data: Value,
+}
+
+pub async fn resolve_federation_entities(ctx: &ContextField<'_>) -> Result<ResolvedValue, Error> {
+    let representations = Vec::<Representation>::deserialize(
+        ctx.param_value_dynamic("representations", crate::resolver_utils::InputResolveMode::Default)?
+            .ok_or_else(|| Error::new("The representations parameter was missing"))?,
+    )
+    .map_err(|error| Error::new(format!("Could not deserialize _Any: {error}")))?;
+
+    let futures = representations.iter().enumerate().map(|(index, representation)| {
+        let mut ctx = ctx.clone();
+        ctx.path.push(index);
+
+        async move {
+            match resolve_representation(&ctx, representation).await {
+                Ok(data) => data,
+                Err(error) => {
+                    ctx.add_error(ctx.set_error_path(error.into_server_error(ctx.item.pos)));
+                    Value::Null
+                }
+            }
+        }
+    });
+
+    Ok(ResolvedValue::new(serde_json::Value::Array(join_all(futures).await)))
+}
+
+async fn resolve_representation(ctx: &ContextField<'_>, representation: &Representation) -> Result<Value, Error> {
+    let entity = ctx
+        .registry()
+        .federation_entities
+        .get(representation.ty.as_str()) // TODO: should this be keyed by NamedType?
+        .ok_or_else(|| Error::new(format!("Unknown __typename in representation: {}", representation.ty)))?;
+
+    let key_being_resolved = entity
+        .find_key(&representation.data)
+        .ok_or_else(|| Error::new("Could not find a matching key for the given representation"))?;
+
+    // The ctx we're passed will have the generic Entity interface in it's type.
+    // But a lot of the resolvers we're going to call into expect to get a context
+    // with a concrete type.  Since we know the __typename here we can help
+    // the resolvers out by passing them a more accurate ResolverContext.
+    // Not sure this is the best idea, but lets find out.
+    let actual_type = ctx.registry().lookup_expecting(&representation.ty).map_err(|error|
+        // This probably indicates a malformed registry, lets return an error
+        Error::new(format!("Internal error: {} could not be looked up in registry: {}", representation.ty, error.message))
+    )?;
+    let resolver_context = ResolverContext::new(&ctx).with_ty(actual_type);
+
+    let data = match key_being_resolved.resolver() {
+        FederationResolver::DynamoUnique => {
```

Comment:
Do we need to extend this every time we add new resolvers for new databases?

Reply:
If we want those to be supported by federation - yes.  I am hoping this glue will be fairly trivial like it is here, but will see how true that is when we add another connector or two.  

We could probably look into some kind of trait/standard resolver/whatever at some point - but it's not yet clear what that'd need to look like (and I didn't want to go refactoring dynamo in particular since we'll be ditching it at some point).

Reply:
Some standard behavior would be nice. Let's figure that out later, I'm just worried that we'll forget this part of the codebase when we implement new connectors...